lib/pipeline.js

```diff
@@ -963,12 +952,16 @@ class PipelineModel extends BaseModel {
                 pipelineId: this.id,
                 type: 'pipeline'
             },
-            sort: 'descending',
-            paginate: {
+            sort: 'descending'
+        };
+
+        // Do not force pagination by default (for backwards compatibility)
+        if (config && config.paginate) {
+            defaultConfig.paginate = {
                 count: PAGINATE_COUNT,
                 page: PAGINATE_PAGE
-            }
-        };
+            };
```

Comment:
Don't think we need this default pagination option here. Base factory will handle the default values.